packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -628,71 +635,70 @@ class RenderConstrainedOverflowBox extends RenderAligningShiftedBox {
+/// error.
 ///
-/// This allows a child to render at the size it would render if it were alone
-/// on an infinite canvas with no constraints. This box will then attempt to
-/// adopt the same size, within the limits of its own constraints. If it ends
-/// up with a different size, it will align the child based on [alignment].
-/// If the box cannot expand enough to accommodate the entire child, the
-/// child will be clipped.
+/// This [RenderBox] sizes its child using a [BoxConstraints] created by
+/// applying [constraintsTransform] to this [RenderBox]'s own [constraints].
+/// This box will then attempt to adopt the same size, within the limits of its
+/// own constraints. If it ends up with a different size, it will align the
+/// child based on [alignment]. If the box cannot expand enough to accommodate
+/// the entire child, the child will be clipped if [clipBehavior] is not
+/// [Clip.none].
 ///
 /// In debug mode, if the child overflows the box, a warning will be printed on
 /// the console, and black and yellow striped areas will appear where the
 /// overflow occurs.
 ///
+/// This [RenderBox] allows the child to selectively enforce some of its
+/// intrinsic sizing rules, partially disregard the constraints set by its
+/// parent render object, and display a warning in debug mode if the parent
+/// render object fails to provide enough space.
+///
 /// See also:
 ///
+///  * [ConstraintsTransformBox], the widget that makes use of this
+///    [RenderObject] and exposes the same functionality.
 ///  * [RenderConstrainedBox], which renders a box which imposes constraints
 ///    on its child.
 ///  * [RenderConstrainedOverflowBox], which renders a box that imposes different
 ///    constraints on its child than it gets from its parent, possibly allowing
 ///    the child to overflow the parent.
-///  * [RenderSizedOverflowBox], a render object that is a specific size but
-///    passes its original constraints through to its child, which it allows to
-///    overflow.
-class RenderUnconstrainedBox extends RenderAligningShiftedBox with DebugOverflowIndicatorMixin {
-  /// Create a render object that sizes itself to the child but does not
-  /// pass the [constraints] down to that child.
+///  * [RenderUnconstrainedBox] which allows its children to render themselves
+///    unconstrained, expands to fit them, and considers overflow to be an error.
+class RenderConstraintsTransformBox extends RenderAligningShiftedBox with DebugOverflowIndicatorMixin {
+  /// Creates a [RenderBox] that sizes itself to the child and modifies the
+  /// [constraints] before passing it down to that child.
   ///
-  /// The [alignment] must not be null.
-  RenderUnconstrainedBox({
+  /// The [alignment] and [clipBehavior] must not be null.
+  RenderConstraintsTransformBox({
     required AlignmentGeometry alignment,
     required TextDirection? textDirection,
-    Axis? constrainedAxis,
+    required BoxConstraintsTransform constraintsTransform,
```

Comment:
Just an idea, let me know what you think:

Make this not required.  If it's null, then it just uses the constraints it has as-is.

Maybe that's kind of pointless, but I was thinking it would allow you to get rid of _unmodified in the main PR: https://github.com/flutter/flutter/pull/64615/files#diff-f5f93c879cef9a102adbf148583de9f3d9a05f9678fdcad7bde106f62200bb34R2606

Reply:
I'll expose the static functions ('unconstrained' etc,.), instead of the named constructors then.